File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.py
class Solution:
    def beautySum(self, s: str) -> int:
        from collections import Counter

        def beauty(sub):
            d = Counter(sub)
            a = list(d.values())
            if len(a) <= 1:
                return 0
            return max(a) - min(a)
        
        n = len(s)
        res = 0
        for i in range(n):
            for j in range(i+1, n+1):
                res += beauty(s[i: j])

        return res

File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    int beautySum(string s) {
        
        int i, j, n=s.size(), sum=0, mini, maxi;

        for(i = 0; i < n; i++) {
            unordered_map<char, int> mpp;
            for(j = i; j < n; j++) {
                mpp[s[j]]++;
                mini = INT_MAX;
                maxi = INT_MIN;

                for(auto item : mpp) {
                    mini = min(mini, item.second);
                    maxi = max(maxi, item.second);
                }

                sum += (maxi - mini);
            }
        }
        return sum;
    }
};